django_sites_extensions/middleware.py

```diff
@@ -27,3 +27,13 @@ def process_request(self, request):
         if redirect_to:
             return redirect(redirect_to, permanent=True)
         return None
+
+
+    def __call__(self, request):
+        """
+        This adds compatibility for django 2.2
+        for more details check 
+        https://docs.djangoproject.com/en/2.0/topics/http/middleware/#upgrading-middleware
+        """
+        response = self.process_request(request)
```

Comment:
Had to upgrade the middleware again as the middleware settings require a call method and the default method of mixin was calling the process_response method and raising an error if it does not exist. While we are not returning any response in case of error and returning none.

Reply:
Can you point me at a case where we hit this error?  At first glance, it seems like the prior code should have worked; returning `None` from `process_request()` should result in the next middleware or the view itself being called, and `MiddlewareMixin` checks for the existence of `process_response()` before trying to call it.

Reply:
My bad, i wanted to write 'get_response' and somehow wrote 'process_response'. We are not setting get_response and as our response is none its called for both django 1.11 and 2.2. 
The mixin calls both.
Code Reference 1.11 : https://github.com/django/django/blob/c494d90c195a739a7298b073eaa6ed987c2fd0bc/django/utils/deprecation.py#L85-L97
Code Reference 2.2
https://github.com/django/django/blob/358973a12eb3105ba084a2d594428a19223b8582/django/utils/deprecation.py#L130-L143

Reply:
Ok, that's supposed to be set by Django when it initializes the middleware by calling `__init__()` (declared in `MiddlewareMixin`).  Is it a problem in the application code, or just in tests that are bypassing the normal initialization process and aren't providing the `get_response` parameter that's now needed?  If the latter, we should fix the tests.